src/hash/sha256.c

```diff
@@ -103,21 +109,36 @@ void sha256_update(sha256_context *ctx, const u8 *input, u32 ilen)
 
 	if ((left > 0) && (remain_ilen >= fill)) {
 		/* Copy data at the end of the buffer */
-		local_memcpy(ctx->sha256_buffer + left, data_ptr, fill);
+		if (((uint32_t)data_ptr >= MRAM_START) && ((uint32_t)data_ptr <=  MRAM_END)) {
+			mram_read((__mram_ptr void *)(uintptr_t)data_ptr, tmp, (fill + (8-(fill%8))));
+			local_memcpy(ctx->sha256_buffer + left, tmp, fill);
```

Comment:
Just wondering, can you use our API `__memcpy_wm(tmp, (__mram_ptr void *)(uintptr_t)data_ptr, fill)`?

Reply:
yes you are right, I will have a try and I'll let you know